mllib/src/main/scala/org/apache/spark/mllib/clustering/StreamingKMeans.scala

```diff
@@ -95,7 +95,7 @@ class StreamingKMeansModel @Since("1.2.0") (
     val discount = timeUnit match {
       case StreamingKMeans.BATCHES => decayFactor
       case StreamingKMeans.POINTS =>
-        val numNewPoints = pointStats.view.map { case (_, (_, n)) =>
+        val numNewPoints = pointStats.iterator.map { case (_, (_, n)) =>
```

Comment:
I guess same question - is the .view needed at all? I'm trying to think of what the original purpose was.